node/bin/src/prover_api/fri_job_manager.rs

```diff
@@ -47,13 +46,30 @@ pub enum SubmitError {
     Other(String),
 }
 
+impl SubmitError {
+    pub fn expected_hash_u32s(&self) -> [u32; 8] {
+        match self {
+            SubmitError::VerificationFailed(expected_hash_u32s, _) => *expected_hash_u32s,
+            _ => panic!("Expected hash u32s not found"),
+        }
+    }
+    pub fn proof_final_register_values(&self) -> [u32; 16] {
+        match self {
+            SubmitError::VerificationFailed(_, proof_final_register_values) => {
+                *proof_final_register_values
+            }
+            _ => panic!("Proof final register values not found"),
+        }
+    }
```

Comment:
I assume you need these because you know that `verify_fri_proof` can only return this variant of `SubmitError`. I would just create a separate error type `FriProofVerificationError` that is wrapped inside `SubmitError`. Then you can return it from `verify_fri_proof` and safely address fields without the need for these methods.